pkg/types/validation/installconfig.go

```diff
@@ -571,3 +578,27 @@ func validateURI(uri string, fldPath *field.Path, schemes []string) field.ErrorL
 	}
 	return field.ErrorList{field.NotSupported(fldPath, parsed.Scheme, schemes)}
 }
+
+// validateIPProxy checks if the given proxy string is an IP and if so checks the service and
+// cluster networks and returns error if the IP belongs in them. Returns nil if the proxy is
+// not an IP address.
+func validateIPProxy(proxy string, n *types.Networking, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	proxyIP := net.ParseIP(proxy)
```

Comment:
You need to get the IP from only the host.
```suggestion
	parsed, err := url.ParseRequestedURI(proxy)
	if err != nil {
		return allErrs
	}
	proxyIP = net.ParseIP(parsed.Host)
```